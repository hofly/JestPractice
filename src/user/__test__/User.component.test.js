import React from 'react';
import { shallow } from 'enzyme'

import User from './../User.component';


describe('Homepage Test', () => {
    it('should render snapshot test for user', () => {
    	let props = {navigation: {state: {params: { userType: 'anonymous' }}}}
    	const wrapper = shallow(<User {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('should simulate & render avatar if avatar is exist ', () => {
    	let props = {navigation: {state: {params: { userType: 'professional' }}}}
    	const wrapper = shallow(<User {...props}/>)
        expect(wrapper.find('Image').exists()).toBe(true)
    })

    it('should render empty when avatar is empty ', () => {
    	let props = {navigation: {state: {params: { userType: 'anonymous' }}}}
    	const wrapper = shallow(<User {...props}/>)
        expect(wrapper.find('Image').exists()).toBe(false)
    })

})