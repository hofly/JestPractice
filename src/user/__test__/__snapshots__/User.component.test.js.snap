// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Test should render snapshot test for user 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingTop": 100,
    }
  }
>
  <Text
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    User
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
     Name: 
    Danny
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
     Age: 
    18
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
     Type: 
    anonymous
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
     Avatar
  </Text>
  <View />
</View>
`;
